prettyconf/loaders.py

```diff
@@ -1,15 +1,44 @@
 # coding: utf-8
 
-import sys
+
 import os
+import sys
+from argparse import Action, ArgumentParser
 from glob import glob
 
+from .exceptions import InvalidConfigurationFile
+
 try:
     from ConfigParser import SafeConfigParser as ConfigParser, NoOptionError, MissingSectionHeaderError
 except ImportError:
     from configparser import ConfigParser, NoOptionError, MissingSectionHeaderError
 
-from .exceptions import InvalidConfigurationFile
+
+class NotSet(str):
+    """
+    A special type that behaves as a replacement for None.
+    We have to put a new default value to know if a variable has been set by
+    the user explicitly. This is useful for the ``CommandLine`` loader, when
+    CLI parsers force you to set a default value, and thus, break the discovery
+    chain.
+    """
+    pass
+
+
+NOT_SET = NotSet()
```

Comment:
It's up to you but you can use `NOT_SET = NotSet("==NOTSET==")` if necessary for serialize something.